frontend/src/helpers/RequireAuth.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { useLocation, Navigate, Outlet } from "react-router-dom";
+
+function RequireAuth() {
+  const localStorageUser = localStorage.getItem("user");
+  let user: any = null;
```

Comment:
I think there should be a typefor user and then We can do a `let user: userType|null = null` in this line and assign in the if statment if needed

Reply:
Fixed. I added a custom type `UserType`.